src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -1815,16 +1815,29 @@ private String determineFilename(
         }
 
         final Date disbursementDate = extractTypeContext.getDisbursementDate();
-        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+        final String filename = buildUniqueFilename(directoryName, finalFormattedCheckFilePrefix, disbursementDate, null);
+
+        LOG.debug("determineFilename(...) - : filename={}", filename);
+        return filename;
+    }
+
+    private String buildUniqueFilename(String directoryName, String finalFormattedCheckFilePrefix, Date disbursementDate, Integer uniquenessSuffix) {
+        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmssSSS", Locale.US);
+
         final String filename =
                 directoryName
                 + File.separator
                 + finalFormattedCheckFilePrefix
                 + "_"
                 + sdf.format(disbursementDate)
+                + (uniquenessSuffix != null ? "-" + uniquenessSuffix : "")
                 + ".xml";
 
-        LOG.debug("determineFilename(...) - : filename={}", filename);
+        if ((new File(filename).exists())) {
+            LOG.info("buildUniqueFilename filename={} already exists, adding uniqueness suffix", filename);
+            return buildUniqueFilename(directoryName, finalFormattedCheckFilePrefix, disbursementDate, ++uniquenessSuffix);
```

Comment:
Your initial call to this method sets `uniquenessSuffix` to null. Will that cause the "++" part on this line to encounter a null pointer?

You may need to make some more adjustments to keep this null-safe.